gammapy/cube/core.py

```diff
@@ -322,30 +320,47 @@ def wcs_pixel_to_skycoord(self, x, y, z):
         energy = Quantity(energy, self.energy_axis.energy.unit)
         return (position, energy)
 
-    def to_sherpa_data3d(self):
-        """
-        Convert sky cube to sherpa `Data3D` object.
+    def to_sherpa_data3d(self, dstype='Data3D'):
         """
-        from .sherpa_ import Data3D
+        Convert sky cube to sherpa `Data3D` or `Data3DInt` object.
 
-        # Energy axes
+        Parameters
+        ----------
+        dstype : {'Data3D', 'Data3DInt'}
+            Sherpa data type.
+        """
+        from .sherpa_ import Data3D, Data3DInt
         energies = self.energies(mode='edges').to("TeV").value
         elo = energies[:-1]
         ehi = energies[1:]
-
-        coordinates = self.sky_image_ref.coordinates()
-        ra = coordinates.data.lon.degree
-        dec = coordinates.data.lat.degree
-
         n_ebins = len(elo)
-        ra_cube = np.tile(ra, (n_ebins, 1, 1))
-        dec_cube = np.tile(dec, (n_ebins, 1, 1))
-        elo_cube = elo.reshape(n_ebins, 1, 1) * np.ones_like(ra)
-        ehi_cube = ehi.reshape(n_ebins, 1, 1) * np.ones_like(ra)
+        if dstype=='Data3DInt':
+            coordinates = self.sky_image_ref.coordinates(mode="edges")
+            ra = coordinates.data.lon
+            dec = coordinates.data.lat
+            ra_cube_hi = np.tile(ra[0:-1, 0:-1].value, (n_ebins, 1, 1))
+            ra_cube_lo = np.tile(ra[0:-1, 1:].value, (n_ebins, 1, 1))
+            dec_cube_hi = np.tile(dec[1:, 0:-1].value, (n_ebins, 1, 1))
+            dec_cube_lo = np.tile(dec[0:-1, 0:-1].value, (n_ebins, 1, 1))
+            elo_cube = elo.reshape(n_ebins, 1, 1) * np.ones_like(ra[0:-1, 0:-1].value) * u.TeV
+            ehi_cube = ehi.reshape(n_ebins, 1, 1) * np.ones_like(ra[0:-1, 0:-1].value) * u.TeV
+            return Data3DInt('', elo_cube.ravel(), ra_cube_lo.ravel(), dec_cube_lo.ravel(), ehi_cube.ravel(),
+                             ra_cube_hi.ravel(), dec_cube_hi.ravel(), self.data.value.ravel(),
+                             self.data.value.shape)
+        if dstype=='Data3D':
+            coordinates = self.sky_image_ref.coordinates()
+            ra = coordinates.data.lon
```

Comment:
@JouvinLea Just a last question: You've removed the `.degree` conversion here. Was that on purpose? Because the unit of `ra` and `dec` is not necessarily in degree, it might only be the case accidentally...